### PR TITLE
Better Timing For Interpolation Calc

### PR DESCRIPTION
**Pull request by JohnAFernandez**
I'll just leave this here....

Before: https://youtu.be/dsdS6d1q93A?t=25

After: https://youtu.be/P8sdYKtmjO8?t=36

Does this by refactoring how the timing is calculated.  It also removes the Next Arrival Time Array, which does not actually do a good job of calculating it, anyway.  We can now rely on the other oo_arrive_time arrays exclusively by subtracting from current time.

There are further improvements not included in this pull request which require a multiplayer version bump, which is forthcoming.

### CODE REVIEW

**Comment by JohnAFernandez** [comment]
That /should/ fix Fred compilation.

File: code/network/multi_obj.cpp [hunk 1]
```diff
@@ -1873,48 +1870,55 @@ void multi_oo_interp(object *objp)
 		return;
 	}	
 
-	// increment his approx "next" time
-	oo_arrive_time_next[objp->instance] += flFrametime;
-
-	// do stream weapon firing for this ship
+	// Now that we know we have a vaild ship, do stream weapon firing for this ship before we do anything else that makes us abort
```
Comment (Goober5000) [inline code/network/multi_obj.cpp]: "valid" is misspelled

File: code/network/multi_obj.cpp [hunk 2]
```diff
@@ -1873,48 +1870,55 @@ void multi_oo_interp(object *objp)
...
 		physics_sim(&objp->pos, &objp->orient, &objp->phys_info, flFrametime);
 		return;
+	// in case they are slightly late, we'll just use the last good point.
+	} else if (t > 1.0f) {
+		t = 1.0f;
 	}	
 
-	// otherwise, blend the two curves together to get the new point
+	// Cyborg17 - we are no longer blending the two curves.  I'm not sure *how*, but they were somehow making the interpolation look 
+	// less eratic when the timing was messed up in the first place.  Now we take the *good* curve at its word.
```
Comment (Goober5000) [inline code/network/multi_obj.cpp]: "erratic" is misspelled

**Comment by JohnAFernandez** [comment]
Comments should be satisfactorily changed.

**Comment by Goober5000** [comment]
Did you push the change?

**Comment by JohnAFernandez** [comment]
Whoops ... nope

**Pull request merged**